### PR TITLE
Fixed a Bug Caused by Clan Creche Aspirants Incorrectly Washing Out of their Creches

### LINKED ISSUE #4193: Error thrown when Clan sibkin reach age 10.
### Environment
Mekhq 0.49.20 nightly 9th June 2024
Widows 10
Java 17.0.11 Adoptium (full re-install prior to logging issue, both JRE and JDK)
2GB RAM allocated, 12GB system

### Description
Upon reaching 10yo, Clan creche personnel throw an error. Daily report states personnel passed/failed trial of position with number of kills. Pass/fail for personnel, and error, is repeated daily until they are sent to academy or re-education (also repeated entry in personnel log). Changing status from student to active also avoids error message.
error
(Composite image, personnel Personel in report, Zale, also finished creche on previous day)

### Files
[CSJ Sibko30360101.cpnx.gz](https://github.com/user-attachments/files/15754257/CSJ.Sibko30360101.cpnx.gz)
Error on my system will occur on next day advance.
Custom rank "Cadet" added at E1. Shouldn't matter? Not a new rank system or file, edited original ranks.xml
[mekhq.log](https://github.com/user-attachments/files/15754271/mekhq.log)
[megamek.log](https://github.com/user-attachments/files/15754273/megamek.log)

### PR DESCRIPTION
**Pull request by IllianiCBT**
This PR fixes a bug in the clan washout logic that didn't correctly prohibit creche aspirants from washing out of daycare. Once the aspirant washed out, they were treated as if they were washing out of a Sibko. This caused a _slew_ of problems.

Anyone using Clan creches in a campaign saved prior to the nightly containing this fix will need to set their aspirants to 'Active' and then re-enroll them.

Unrelated to Clan Creches, I also expanded the qualifications offered by Clan Reeducation Camps to include Warrior Caste options, as there are Clans which embrace warriors from outside of Clan culture; so seemed appropriate that those qualifications should be offered, too.

## Closes
Closes #4193

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/education/EducationController.java [hunk 1]
```diff
@@ -454,11 +454,11 @@ private static boolean ongoingEducation(Campaign campaign, Person person, Academ
      * @param person the person to determine graduation for
      * @param academy the academy to determine graduation from
      * @param resources the resources to use for graduation
-     * @return true if the person successfully graduates; otherwise, false
+     * @return true, if the person successfully graduates; otherwise, false
      */
     private static boolean graduationPicker(Campaign campaign, Person person, Academy academy, ResourceBundle resources) {
         if ((academy.isClan()) && (academy.isPrepSchool())) {
-            if (person.getAge(campaign.getLocalDate()) < 10) {
+            if (person.getAge(campaign.getLocalDate()) == 10) {
```
Comment (pavelbraginskiy) [inline MekHQ/src/mekhq/campaign/personnel/education/EducationController.java:461]: Is this right? Wouldn't this make it call graduateClanSibko on those below age 10?
If that _is_ right, I think the logic is unintuitive enough to warrant an extra comment.

Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/personnel/education/EducationController.java:461]: Persons under the age of 10 shouldn't be graduating. However, as the user can force a graduation that'll cause problems. I'll go ahead and add an extra clause. Thanks for pointing this out!

Comment (IllianiCBT) [reply MekHQ/src/mekhq/campaign/personnel/education/EducationController.java]: Fixed :)

**Comment by IllianiCBT** [comment]
Flipping this to Draft so I can add some more bug fixes.

**Comment by IllianiCBT** [comment]
Closing as no longer relevant. Clan Education is being removed entirely before 49.20's release and reworked for 50.1

**Pull request closed without merging**